formsOnline/formsOnline-war/src/main/java/org/silverpeas/components/formsonline/control/FormsOnlineSessionController.java

```diff
@@ -287,19 +321,166 @@ public RequestsByStatus getAllUserRequests() throws FormsOnlineDatabaseException
   }
 
   public RequestsByStatus getHomepageValidatorRequests() throws FormsOnlineDatabaseException {
-    return getService().getValidatorRequests(getComponentId(), isWorkgroupEnabled(), getUserId(),
+    return getService().getValidatorRequests(getRequestsFilter(), getUserId(),
         new PaginationPage(1, DEFAULT_ITEM_PER_PAGE));
   }
 
   public RequestsByStatus getAllValidatorRequests() throws FormsOnlineDatabaseException {
-    return getService()
-        .getValidatorRequests(getComponentId(), isWorkgroupEnabled(), getUserId(), null);
+    RequestsFilter filter = getRequestsFilter();
+    if (getCurrentForm() != null) {
+      filter.getFormIds().add(Integer.toString(getCurrentForm().getId()));
+    }
+    return getService().getValidatorRequests(filter, getUserId(), null);
+  }
+
+  public Form getCurrentEmptyForm() throws SilverpeasException {
+    try {
+      PublicationTemplate pubTemplate = getCurrentPublicationTemplate(true);
+      return getEmptyForm(pubTemplate);
+    } catch (Exception e) {
+      throw new SilverpeasException("Can't load form "+getCurrentForm().getXmlFormName(), e);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Can't load form " 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1192)